src/main/serialportMap.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import { AutoDetectTypes } from '@serialport/bindings-cpp';
+import { WebContents } from 'electron';
+import { SerialPort, SerialPortOpenOptions } from 'serialport';
+
+export type Renderer = Pick<WebContents, 'on' | 'send' | 'id'>;
+export type OpenPort = {
+    serialPort: SerialPort;
+    renderers: Renderer[];
+    settingsLocked: boolean;
+    opening: boolean;
+    options: SerialPortOpenOptions<AutoDetectTypes>;
+};
+
+const mapPath = (path: string) =>
+    process.platform === 'win32' ? path.toLowerCase() : path;
+
+class SerialPortMap {
```

Comment:
It's mainly because I originally wrote it using by extending `Map`, and then we discussed using _Delegation_ instead.  We could, of course, rewrite it in order to avoid using class, if that is preferred?

Reply:
It still has many similarities with the build in `Map` type, so I think it is also fine to keep it like this. I see few advantages in turning it into a functional style, where you pass the object as a parameter on every call instead (If I understood you correctly).